bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/ArithmeticGroupFunction.java

```diff
@@ -221,14 +221,31 @@ public State calculate(Set<Item> items) {
          */
         @Override
         public State getStateAs(Set<Item> items, Class<? extends State> stateClass) {
-            final State state = calculate(items);
+            State state = calculate(items);
             if (stateClass.isInstance(state)) {
                 return state;
             } else {
-                return UnDefType.UNDEF;
+                if (stateClass == DecimalType.class) {

```

Comment:
I think this check makes sense for Or: Interpreted as OnOffType its state will be On if _at least_ one member is active, Off otherwise.  Counting active members is strictly more information but has the same meaning when converted to OnOffType.

For Xor, I don’t quite see it as sensible.  The groups is On if _exactly_ one member is active, Off otherwise.  But this would again count the active members, and converting _that_ to OnOffType would again result in On if _at least_ one member was active.

Do we have a use case where we make use of this feature?  Because otherwise I’d recommend removing it until we do, if only to avoid these possibly confusing conversions.

Reply:
I fully agree with what you wrote. The docs however let assume another use case for this. On "And" the return will be how many states are not active and thus this information is not used to determine a state but to get information additional to the state returned. For "XOR" that would mean information on how many active states exist on the items to see why for example the state is in a passive state.

Reply:
Good point, let’s keep the conversion then.